core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -642,10 +642,10 @@ class SnappyParser(session: SnappySession)
         ) |
         MATCH ~> UnresolvedAttribute.quoted _
     ) |
-    '{' ~ FN ~ ws ~ identifier ~ '(' ~ (expression * commaSep) ~ ')' ~ ws ~ '}' ~ ws ~> {
-      (f: Any, e: Any) =>
-        f.asInstanceOf[String] match {
-          case f if f.equalsIgnoreCase("TIMESTAMPADD") =>
+    '{' ~ FN ~ ws ~ functionIdentifier ~ '(' ~ (expression * commaSep) ~ ')' ~ ws ~ '}' ~ ws ~> {
+      (f: FunctionIdentifier, e: Any) =>
+        f match {
+          case f if f.funcName.equalsIgnoreCase("TIMESTAMPADD") =>
```

Comment:
@soubhik-c Why is this special treatment of TIMESTAMPADD and SQL_TSI_DAY? Can't it be handled like a regular function call with any checks for the arguments inside the implementation of TimestampAdd. I guess you will need to add a new function which will internally call DateAdd but that is much better than cluttering the parser. In future too if we add more then folks will simply clutter the parser even more instead of cleanly handling it in a separate function.

The COUNT case is the one and only special case which can accept COUNT(*) arguments while no other function can, hence the special handling for COUNT else ideally even that should be avoided.